projects/igniteui-angular/src/lib/grid/grid-toolbar.spec.ts

```diff
@@ -432,48 +317,59 @@ describe('IgxGrid - Grid Toolbar', () => {
     });
 
     it('shows Column Hiding button with default content when columnHiding=true.', () => {
-        const fixture = TestBed.createComponent(GridToolbarTestPage1Component);
-        const grid = fixture.componentInstance.grid1;
         grid.showToolbar = true;
         grid.columnHiding = true;
         fixture.detectChanges();
         expect(grid.toolbar.columnHidingUI).toBeDefined();
 
-        const button = fixture.debugElement.queryAll(By.css('button')).find((b) => b.nativeElement.name === 'btnColumnChooser');
+        const button = fixture.debugElement.queryAll(By.css('button')).find((b) => b.nativeElement.name === 'btnColumnHiding');
         expect(button).toBeDefined();
         const btnText = button.nativeElement.innerText.toLowerCase();
         expect(btnText.includes('0') && btnText.includes('visibility') && !btnText.includes('visibility_off')).toBe(true);
     });
 
     it('toggleColumnHidingUI() method opens and closes the ColumnHiding dropdown.', () => {
-        const fix = TestBed.createComponent(GridToolbarTestPage1Component);
-        const grid = fix.componentInstance.grid1;
         grid.showToolbar = true;
         grid.columnHiding = true;
-        fix.detectChanges();
+        fixture.detectChanges();
 
-        const dropdown = fix.debugElement.query(By.css('igx-drop-down'));
-        const dropDownDiv = dropdown.query(By.css('div.igx-drop-down__list'));
+        getColumnHidingButton().nativeElement.click();
 
-        expect(JSON.stringify(dropDownDiv.classes)).toBe('{"igx-toggle--hidden":true,"igx-toggle":false}');
-        expect(dropDownDiv.query(By.css('igx-column-hiding'))).toBe(null);
+        const dropDownDiv = getOverlay();
+        expect(dropDownDiv).not.toBe(null);
+        expect(dropDownDiv.querySelector('igx-column-hiding')).not.toBe(null);
 
-        const button = fix.debugElement.queryAll(By.css('button')).find((b) => b.nativeElement.name === 'btnColumnChooser').nativeElement;
-        grid.toolbar.toggleColumnHidingUI();
-        // button.click();
+        getColumnHidingButton().nativeElement.click();
 
-        expect(JSON.stringify(dropDownDiv.classes)).toBe('{"igx-toggle--hidden":false,"igx-toggle":true}');
-        expect(dropDownDiv.query(By.css('igx-column-hiding'))).not.toBe(null);
-        grid.toolbar.columnHidingDropdown.close();
-        // grid.toolbar.toggleColumnHidingUI();
-        // expect(dropDownDiv.query(By.css('igx-column-hiding'))).toBe(null);
-        // expect(JSON.stringify(dropDownDiv.classes)).toBe('{"igx-toggle--hidden":true,"igx-toggle":false}');
     });
+
+    function getToolbar() {
+        return fixture.debugElement.query(By.css('igx-grid-toolbar'));
+    }
+
+    function getOverlay() {
+        const div = fixture.debugElement.nativeElement.parentElement.lastChild;
```

Comment:
It is more clear if you take the overlay div like this:
const div = document.getElementsByClassName('igx-overlay')[0];

Reply:
It introduces other issues in the tests and involves other changes so I will leave this as is.